src/index.ts

```diff
@@ -0,0 +1,130 @@
+import * as path from "path"
+import * as fs from "@ts-common/fs"
+import * as md from "@ts-common/commonmark-to-markdown"
+import * as azureMd from "@azure/openapi-markdown"
+import * as ai from "@ts-common/async-iterator"
+import * as jp from "@ts-common/json-parser"
+import * as it from "@ts-common/iterator"
+import * as json from "@ts-common/json"
+import * as sm from "@ts-common/string-map"
+
+export const cli = async <T>(f: (path: string) => AsyncIterable<T>): Promise<number> => {
+  try {
+    const errors = await f("./")
+    let errorsNumber = 0
+    for await (const e of errors) {
+      console.error(e)
+      ++errorsNumber
+    }
+    console.log(`errors: ${errorsNumber}`)
+    return errorsNumber === 0 ? 0 : 1
+  } catch (e) {
+    console.error("INTERNAL ERROR")
+    console.error(e);
+    return 1
+  }
+}
+
+export type JsonParseError = {
+  readonly code: "JSON_PARSE",
+  readonly error: jp.ParseError
+}
+
+export type FileError = {
+  code: "NO_OPEN_API_FILE_FOUND" | "UNREFERENCED_OPEN_API_FILE"
+  readonly message: string
+  readonly readMeUrl: string
+  readonly openApiUrl: string
+}
+
+export type Error = JsonParseError | FileError
+
+export const avocado = (dir: string): ai.AsyncIterableEx<Error> =>
```

Comment:
should we have a parameter to indicate which actions/rules to perform on the dir? Eventhough the tool is focused on the directory and file structure, as it grows I wonder why we would have to run all the criteria at once. Are you thinking of the tool running different "checks"?

Reply:
We may have it in the future but I don't want to introduce dummy parameters for now